File: Data Structure/String/Medium/LET_no848_Shifting Letters.cpp
class Solution {
public:
    string shiftingLetters(string s, vector<int>& shifts) {
        int pre = 0, cur;
        for (int i = s.size() - 1; i >= 0; i--) {
            cur = (pre + shifts[i]) % 26;
            pre = cur;
            s[i] = 'a' + (s[i] - 'a' + cur) % 26;
        }
        return s;
    }
};

File: Data Structure/String/Medium/LET_no848_Shifting Letters.py
class Solution:
    def shiftingLetters(self, s: str, shifts: List[int]) -> str:
        s = list(s)
        pre = 0
        for i in range(len(shifts) - 1, -1, -1):
            cur = pre + shifts[i]
            pre = cur
            s[i] = chr(ord('a') + (ord(s[i]) - ord('a') + cur) % 26)
        return ''.join(s)